test/e2e/operator_test.go

```diff
@@ -1,26 +1,316 @@
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Authorization", bearer)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("curl command failed: %v\n%v", err, resp)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	return body, nil
+}
+
+// runQuery is similar to runQueries that lives in origin/test/extended/prometheus
+func runQuery(t *testing.T, kclient *kubernetes.Clientset, query, ns, execPodName, baseURL, bearerToken string, expected bool) error {
+	// expect all correct metrics within a reasonable time period
+	var err error
+	for i := 0; i < waitForPrometheusStartSeconds; i++ {
+		//TODO when the http/query apis discussed at https://github.com/prometheus/client_golang#client-for-the-prometheus-http-api
+		// and introduced at https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api.go are vendored into
+		// openshift/origin, look to replace this homegrown http request / query param with that API
+		contents, err := getBearerTokenURLViaPod(t, kclient, ns, execPodName, fmt.Sprintf("%s/api/v1/query?%s", baseURL, (url.Values{"query": []string{query}}).Encode()), bearerToken)
+		if err != nil {
+			return err
+		}
+		result := prometheusResponse{}
+		json.Unmarshal(contents, &result)
+		metrics := result.Data.Result
+
+		if (len(metrics) > 0 && !expected) || (len(metrics) == 0 && expected) {
+			err = fmt.Errorf("promQL query: %s had reported incorrect results: %v", query, metrics)
+		}
+
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Second)
+	}
+	return err
+}
+
+func tlsConfig(t *testing.T, cert, key []byte) (*tls.Config, error) {
+	certFile, err := ioutil.TempFile(os.TempDir(), "testtlscert-")
+	if err != nil {
+		return nil, err
+	}
+	keyFile, err := ioutil.TempFile(os.TempDir(), "testtlskey-")
+	if err != nil {
+		return nil, err
+	}
+	if _, err = certFile.Write(cert); err != nil {
+		return nil, err
+	}
+
+	if err := certFile.Close(); err != nil {
+		return nil, err
+	}
+	if _, err = keyFile.Write(key); err != nil {
+		return nil, err
+	}
+
+	if err := keyFile.Close(); err != nil {
+		return nil, err
+	}
+
+	tlscert, err := tls.LoadX509KeyPair(certFile.Name(), keyFile.Name())
+	if err != nil {
+		return nil, fmt.Errorf("unable to use specified client cert (%s) & key (%s): %s", certFile.Name(), keyFile.Name(), err)
+	}
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+	tlsConfig.Certificates = []tls.Certificate{tlscert}
+	tlsConfig.BuildNameToCertificate()
+	return tlsConfig, nil
+}
+
+func createTestNamespace(client *kubernetes.Clientset, namespaceName string) (*corev1.Namespace, error) {
+	ns, err := client.CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	})
+	return ns, err
+}
+
+func cleanupTestObjects(client *kubernetes.Clientset, ns string) {
+	client.CoreV1().Namespaces().Delete(ns, &metav1.DeleteOptions{})
+}
+
+// took this from service-ca-operator test
```

Comment:
Suggesting using uuid instead:

https://github.com/openshift/cluster-kube-controller-manager-operator/blob/master/vendor/github.com/openshift/library-go/pkg/config/leaderelection/leaderelection.go#L32

Reply:
But this isn't really the best way to deal with ensuring name uniqueness in tests. Setting `ObjectMeta.GenerateName` is likely preferable if it's possible to create the object before knowing the name:

https://github.com/kubernetes-sigs/kubefed/blob/master/test/e2e/framework/framework.go#L308

It's also possible to directly call the apiserver's name generating logic:

https://github.com/kubernetes-sigs/kubefed/blob/master/test/e2e/defaulting.go#L66

Reply:
looking into it...

Reply:
ah, I knew there must be a better way.  fixed.